Source/portal.cpp

```diff
@@ -10,203 +10,174 @@ int WarpDropY[MAXPORTAL] = { 40, 40, 40, 40 };
 
 void __cdecl InitPortals()
 {
-	int i; // edi
+    int i;
 
-	for(i = 0; i < MAXPORTAL; i++)
-	{
-		if(delta_portal_inited(i))
-			portal[i].open = 0;
-	}
+    for (i = 0; i < MAXPORTAL; i++) {
+        if (delta_portal_inited(i))
+            portal[i].open = FALSE;
+    }
 }
 
-void __fastcall SetPortalStats(int i, int o, int x, int y, int lvl, int lvltype)
+void __fastcall SetPortalStats(int i, BOOL o, int x, int y, int lvl, int lvltype)
 {
-	portal[i].x = x;
-	portal[i].setlvl = 0;
-	portal[i].y = y;
-	portal[i].open = o;
-	portal[i].level = lvl;
-	portal[i].ltype = lvltype;
+    portal[i].x = x;
+    portal[i].setlvl = FALSE;
+    portal[i].y = y;
+    portal[i].open = o;
+    portal[i].level = lvl;
+    portal[i].ltype = lvltype;
 }
 
 void __fastcall AddWarpMissile(int i, int x, int y)
 {
-	int mi; // eax
+    int mi;
 
-	missiledata[MIS_TOWN].mlSFX = -1;
-	dMissile[x][y] = 0;
-	mi = AddMissile(0, 0, x, y, 0, MIS_TOWN, 0, i, 0, 0);
+    missiledata[MIS_TOWN].mlSFX = -1;
+    dMissile[x][y] = 0;
+    mi = AddMissile(0, 0, x, y, 0, MIS_TOWN, 0, i, 0, 0);
 
-	if ( mi != -1 )
-	{
-		SetMissDir(mi, 1);
+    if (mi != -1) {
+        SetMissDir(mi, 1);
 
-		if ( currlevel )
-			missile[mi]._mlid = AddLight(missile[mi]._mix, missile[mi]._miy, 15);
+        if (currlevel)
+            missile[mi]._mlid = AddLight(missile[mi]._mix, missile[mi]._miy, 15);
 
-		missiledata[MIS_TOWN].mlSFX = LS_SENTINEL;
-	}
+        missiledata[MIS_TOWN].mlSFX = LS_SENTINEL;
+    }
 }
 
 void __cdecl SyncPortals()
 {
-	int v0; // edi
-	int *v1; // esi
-	int v2; // eax
-
-	v0 = 0;
-	v1 = &portal[0].level;
-	do
-	{
-		if ( *(v1 - 3) )
-		{
-			if ( currlevel )
-			{
-				v2 = currlevel;
-				if ( setlevel )
-					v2 = (unsigned char)setlvlnum;
-				if ( *v1 == v2 )
-					AddWarpMissile(v0, *(v1 - 2), *(v1 - 1));
-			}
-			else
-			{
-				AddWarpMissile(v0, WarpDropX[v0], WarpDropY[v0]);
-			}
-		}
-		v1 += 6;
-		++v0;
-	}
-	while ( (signed int)v1 < (signed int)&portal[MAXPORTAL].level );
+    int i;
+
+    for (i = 0; i < MAXPORTAL; i++) {
+        if (!portal[i].open)
+            continue;
+        if (!currlevel)
+            AddWarpMissile(i, WarpDropX[i], WarpDropY[i]);
+        else {
+            int lvl = currlevel;
+            if (setlevel)
+                lvl = (unsigned char)setlvlnum;

```

Comment:
Yes, setlvlnum is a signed char.

Reply:
Looked in to it and I'm pretty sure setlvlnum should have been BYTE, so I'm going to correct that, but tell me if you think I'm wrong here.

Reply:
That's what it looked like to me, I just didn't want to include all the fixes for that in this PR (assuming it would be done whenever the file that setlvlnum comes from gets cleaned).

Reply:
Here's the commit where I changed it https://github.com/diasurgical/devilution/commit/f1c8439c64733a166e88dfa5487e1baee7927548

I guess there can be a balance between what is to much for one PR, in this case I feel including it would have been fine.

Reply:
For the time being I think it's best to keep PRs as brief as possible (touch only files that are necessary) to avoid causing conflicts with other PRs in the queue.